src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -478,3 +478,101 @@ function CircularLadderGraph(n::T) where {T <: Integer}
     add_edge!(g, n+1, 2*n)
     return g
 end
+
+"""
+    BarbellGraph(n)
+
+Create a [barbell graph](https://en.wikipedia.org/wiki/Barbell_graph) consisting of a clique of size `n1` connected by an edge to a clique of size `n2`.
+
+### Implementation Notes
+Preserves the eltype of `n1` and `n2`. Will error if the required number of vertices
+exceeds the eltype.
+`n1` and `n2` must be at least 1 so that both cliques are non-empty.
+"""
```

Comment:
Maybe add some comment here, on which vertices belong to which clique and what are the vertices that connect them. By doing that, we guarantee to the user, that we do not suddenly change the order of the vertices, so that his code will break. You could also add a test case for that.